applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -107,4 +126,94 @@ impl wallet_server::Wallet for WalletGrpcServer {
 
         Ok(Response::new(TransferResponse { results }))
     }
+
+    async fn get_transaction_info(
+        &self,
+        request: Request<GetTransactionInfoRequest>,
+    ) -> Result<Response<GetTransactionInfoResponse>, Status>
+    {
+        let message = request.into_inner();
+
+        let queries = message.transaction_ids.into_iter().map(|tx_id| {
+            let mut transaction_service = self.get_transaction_service();
+            async move { transaction_service.get_any_transaction(tx_id).await }
+        });
+
+        let transactions = future::try_join_all(queries)
+            .await
+            .map_err(|err| Status::unknown(err.to_string()))
+            .and_then(|transactions| {
+                // If any of the transaction IDs are not known, this call fails
+                if let Some(pos) = transactions.iter().position(Option::is_none) {
+                    return Err(Status::not_found(format!(
+                        "Transaction ID at position {} not found",
+                        pos
+                    )));
+                }
+                Ok(transactions.into_iter().map(Option::unwrap))
+            })?;
+
+        let wallet_pk = self.wallet.comms.node_identity_ref().public_key();
+
+        let statuses = transactions
+            .into_iter()
+            .map(|tx| convert_wallet_transaction_into_transaction_info(tx, wallet_pk))
+            .collect();
+
+        Ok(Response::new(GetTransactionInfoResponse { statuses }))
+    }
+}
+
+fn convert_wallet_transaction_into_transaction_info(
+    tx: models::WalletTransaction,
+    wallet_pk: &CommsPublicKey,
+) -> TransactionInfo
+{
+    use models::WalletTransaction::*;
+    match tx {
+        PendingInbound(tx) => TransactionInfo {
+            tx_id: tx.tx_id,
+            source_pk: tx.source_public_key.to_vec(),
```

Comment:
I think this returned value should be in hex, so that a human user can see, verify and correlate easily with their console wallet.

Reply:
see here https://github.com/tari-project/tari/pull/2553#discussion_r563550766